src/otx/core/data/dataset/anomaly/dataset.py

```diff
@@ -0,0 +1,151 @@
+            # Note: this part of code is brittle. Ideally Datumaro should return masks
+            # Another major problem with this is that it assumes that the dataset passed is in MVTec format
+            mask_file_path = (
+                Path("/".join(datumaro_item.media.path.split("/")[:-3]))
+                / "ground_truth"
+                / f"{('/'.join(datumaro_item.media.path.split('/')[-2:])).replace('.png','_mask.png')}"
+            )
+            mask = torch.zeros(1, img_shape[0], img_shape[1], dtype=torch.uint8)
+            if mask_file_path.exists():
+                # read and convert to binary mask
+                mask = (io.read_image(str(mask_file_path), mode=io.ImageReadMode.GRAY) / 255).to(torch.uint8)
+            item = AnomalySegmentationDataItem(
+                image=img_data,
+                img_info=ImageInfo(
+                    img_idx=index,
+                    img_shape=img_shape,
+                    ori_shape=img_shape,
+                    image_color_channel=self.image_color_channel,
+                ),
+                label=label,
+                mask=Mask(mask),
+            )
+        elif self.task_type == OTXTaskType.ANOMALY_DETECTION:
+            # Note: this part of code is brittle. Ideally Datumaro should return masks
+            mask_file_path = (
+                Path("/".join(datumaro_item.media.path.split("/")[:-3]))
+                / "ground_truth"
+                / f"{('/'.join(datumaro_item.media.path.split('/')[-2:])).replace('.png','_mask.png')}"
+            )
+            mask = torch.zeros(1, img_shape[0], img_shape[1], dtype=torch.uint8)
+            if mask_file_path.exists():
+                # read and convert to binary mask
+                mask = (io.read_image(str(mask_file_path), mode=io.ImageReadMode.GRAY) / 255).to(torch.uint8)
+            boxes, _ = masks_to_boxes(mask)
+            item = AnomalyDetectionDataItem(
+                image=img_data,
+                img_info=ImageInfo(
+                    img_idx=index,
+                    img_shape=img_shape,
+                    ori_shape=img_shape,
+                    image_color_channel=self.image_color_channel,
+                ),
+                label=label,
+                boxes=BoundingBoxes(boxes[0], format=BoundingBoxFormat.XYXY, canvas_size=img_shape),
+                # mask is used for pixel-level metric computation. We can't assume that this will always be available
+                mask=Mask(mask),
+            )
+        else:
+            msg = f"Task {self.task_type} is not supported yet."
+            raise NotImplementedError(msg)
+
+        # without ignore the following error is returned
+        # Incompatible return value type (got "Any | None", expected
+        # "AnomalyClassificationDataItem | AnomalySegmentationDataBatch | AnomalyDetectionDataBatch")
+        return self._apply_transforms(item)  # type: ignore[return-value]
+
+    @property
+    def collate_fn(self) -> Callable:
```

Comment:
@djdameln, would be good to get your opinion here. Would you prefer having a single `AnomalyDataset` that returns the return_val based on the task, or should we have a dataset for each task?